app/importers/importers/importer.rb

```diff
@@ -22,63 +22,104 @@ class Importer
       @metadata = metadata
       @errors ||= []
+      @conflicting_item_was_kept = false
+      explanation = 'options[:keep_conflicting_items] should be true, false or nil.'
+      raise ArgumentError, explanation unless [true,false,nil].include? options[:keep_conflicting_items]
+      @keep_conflicting_items = options[:keep_conflicting_items].nil? ? true : options[:keep_conflicting_items]
+    end
+
+    # This is the only method called on this class
+    # by the rake task after instantiation.
+    # It reads metadata from file, creates or finds
+    # an item based on it, blanks out some of its metadata for freshening up from import if needed,
+    # then saves it to the database.
+    def import
+      self.class.without_auto_timestamps do
+        if preexisting_item?
+          blank_out_for_reimport(target_item)
+        end
+        return if @conflicting_item_was_kept
+        common_populate
+        populate
+        save_target_item
+      end
+    end
+
+    # Is there a preexisting item with the same friendlier_id *and* the same type?
+    def preexisting_item?
+      # Calling target_item:
+      #   * checks for any conflicting items
+      #       * deletes them if it is allowed to,
+      #       * sets @conflicting_item_was_kept otherwise.
+      #   * creates or sets @target_item (without blanking it out -- that's done in the import method.)
+      #   * sets boolean ivar @preexisting_item_was_found
+      target_item
+      # If a conflicting item still exists, return false.
+      return false if @conflicting_item_was_kept
+      @preexisting_item_was_found
     end
 
 
     # Returns an ActiveRecord model instance, that is the primary target item to save in the import.
-    # * Will find an existing object in the db, and blank out some of it's metadata for freshening up from import
-    # * Or a new AR model if it didn't already exist.
+    # * Will find an existing object in the db if it exists,
+    # * or a new AR model if it didn't already exist.
     #
     # Lazily loads with memoization, so checks the db on first time you call it, after that remembers
     # the object it came up with before, so you can call it as much as you want.
     #
     # Often used in #populate for setting metadata based on import json files:
     #     target_item.title = "whataever"
     #
-    # Also sets ivar @preexisting_item for use with `#preexisting_item?` as a side-effect (bit hacky,
+    # Also sets ivar @preexisting_item_was_found for use with `#preexisting_item?` as a side-effect (bit hacky,
     # but we're refactoring here.)
+    #
+    # If there are conflicting items, either removes them or sets @conflicting_item_was_kept.
     def target_item
       @target_item ||= begin
-        model_object = self.class.destination_class.where(friendlier_id:@metadata['id']).first
-        if (model_object)
-          @preexisting_item = true
+        items_with_same_id = check_for_items_with_same_id
+
```

Comment:
This is also really confusing, what's going to be returned here, what we can expect. It's a hash? Which has one key, either :conflicting, or :preexisting?  And the value is... a model?  Or sometimes it returns nil? Or an empty hash?

I can figure this out following it all through the code, but it's a baroque API.